### PR TITLE
Fixed the Double elevation issue in My Events section

### LINKED ISSUE #1173: Double elevation in tabs and action bar
### Bug Report    

**Actual Behaviour**
Double elevation is present in tabs and action bar of the app which needs to be removed.

**Android version and Phone Model**
Android 6.0.1 and Redmi Note 3

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by tavishjain**
Fixes #1173 

Checklist:

- [1 ] I have checked for PMD and check-style issues 
- [1 ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [1 ] My branch is up-to-date with the Upstream development branch.

Changes: Changed the layout in event_list_fragment.xml to fix the double elevation issue. 

Screenshots for the change:

Previous : 
whatsapp image 2019-01-01 at 3 20 21 pm 1

Current : 

whatsapp image 2019-01-01 at 3 20 21 pm

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -18,7 +18,7 @@ android {
     }
 
     compileSdkVersion versions.compileSdk
-    buildToolsVersion versions.buildTools
+    buildToolsVersion '28.0.3'
```
Comment (ShridharGoel) [inline app/build.gradle]: Should not change this file.

File: build.gradle [hunk 2]
```diff
@@ -7,7 +7,7 @@ buildscript {
         google()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.1.4'
+        classpath 'com.android.tools.build:gradle:3.2.1'
```
Comment (ShridharGoel) [inline build.gradle]: Should not change this file.

**Comment by iamareebjamal** [comment]
This is not correct. Instead, the shadow from action bar should be gone, not the tabs

**Comment by tavishjain** [comment]
We would have to change the theme to NoActionBar and then use a custom toolbar with TabLayout and then setting it as the support action bar to get the desired effect.

**Comment by iamareebjamal** [comment]
Sure, go ahead

**Comment by iamareebjamal** [comment]
Custom Toolbar won't be needed. Only a theme application would be sufficient

**Comment by tavishjain** [comment]
@iamareebjamal but how would you set the theme for a fragment ?

**Comment by iamareebjamal** [comment]
You'll set the theme for the toolbar. I think toolbar is still being used?

**Comment by tavishjain** [comment]
@iamareebjamal I have added the toolbar to do this fix. It wasn't added previously

**Comment by iamareebjamal** [comment]
OK

**Comment by iamareebjamal** [comment]
> This is not correct. Instead, the shadow from action bar should be gone, not the tabs

This is not the fix of the issue nor the desired behaviour

**Comment by tavishjain** [comment]
@iamareebjamal I guess this is fine now 

whatsapp image 2019-01-01 at 10 29 29 pm

**Comment by iamareebjamal** [comment]
Show screenshots of other fragments

**Comment by tavishjain** [comment]
Just a  minute @iamareebjamal

**Comment by tavishjain** [comment]
whatsapp image 2019-01-01 at 11 21 31 pm
whatsapp image 2019-01-01 at 11 20 02 pm 5
whatsapp image 2019-01-01 at 11 20 02 pm 4
whatsapp image 2019-01-01 at 11 20 02 pm 3
whatsapp image 2019-01-01 at 11 20 02 pm 2
whatsapp image 2019-01-01 at 11 20 02 pm 1
whatsapp image 2019-01-01 at 11 20 02 pm

**Comment by tavishjain** [comment]
Are they fine @iamareebjamal ?

**Comment by iamareebjamal** [comment]
No, we want shadows in other fragments

**Comment by tavishjain** [comment]
Is this fine now ? @iamareebjamal 

merge_from_ofoct

**Comment by tavishjain** [comment]
@iamareebjamal ?

**Comment by tavishjain** [comment]
@iamareebjamal can you please verify this now ?
@ShridharGoel I guess @iamareebjamal isn't available today. Could you verify this ?

**Comment by iamareebjamal** [comment]
Move dpToPx to utis and rather than using onCreateView, use onStart

**Comment by iamareebjamal** [comment]
And share screenshots of other fragments

**Comment by tavishjain** [comment]
@iamareebjamal here are the final screenshots of all the fragments, merged into one single image

merge_from_ofoct

File: app/src/main/java/com/eventyay/organizer/core/event/list/EventListFragment.java [hunk 3]
```diff
@@ -59,6 +64,8 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         // Inflate the layout for this fragment
         binding = DataBindingUtil.inflate(inflater, R.layout.event_list_fragment, container, false);
 
+        appBarLayout = getActivity().findViewById(R.id.app_bar_main);
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/event/list/EventListFragment.java]: You already have the binding, no need of this

File: app/src/main/java/com/eventyay/organizer/core/event/list/EventListFragment.java [hunk 4]
```diff
@@ -122,6 +134,14 @@ public void onStop() {
         refreshLayout.setOnRefreshListener(null);
     }
 
+    @Override
+    public void onDestroy() {
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/event/list/EventListFragment.java]: onStop

File: app/src/main/java/com/eventyay/organizer/core/event/list/EventListFragment.java (hunk 3 above)
Comment (tavishjain) [reply app/src/main/java/com/eventyay/organizer/core/event/list/EventListFragment.java]: @iamareebjamal data binding would let me access the layout elements that are present in the inflated layout and not the app bar layout. Hence this needs to be done.

**Comment by kush-mish** [comment]
@tavishjain please check the login fragment too. It was concerning when I was working on this issue.

File: app/src/main/java/com/eventyay/organizer/core/event/list/EventListFragment.java (hunk 4 above)
Comment (tavishjain) [reply app/src/main/java/com/eventyay/organizer/core/event/list/EventListFragment.java]: @iamareebjamal check the latest commit

**Comment by tavishjain** [comment]
> @tavishjain please check the login fragment too. It was concerning when I was working on this issue.

What part in the login fragment ?

**Comment by kush-mish** [comment]
The shadow below the action bar. Since it's not there in the image you've provided.

**Comment by tavishjain** [comment]
@kush-mish could you please tell me the which image are you refering ?

**Comment by kush-mish** [comment]
> @kush-mish could you please tell me the which image are you referring?

You've not provided the screenshot of loginFragment, in any of your images.(have I missed it?)

**Comment by tavishjain** [comment]
@kush-mish this is a PR for the double elevation issue in the my events fragment, which is solved I guess. If you are referring to some other issue, please state which you are talking about

**Comment by tavishjain** [comment]
@iamareebjamal any more changes required ?

File: app/build.gradle [hunk 5]
```diff
@@ -180,7 +180,7 @@ dependencies {
     implementation 'me.xdrop:fuzzywuzzy:1.1.9'
 
     // Glide
-    implementation ("com.github.florent37:glidepalette:2.1.2") {
+    implementation("com.github.florent37:glidepalette:2.1.2") {
```
Comment (iamareebjamal) [inline app/build.gradle]: Unneeded

File: app/src/main/java/com/eventyay/organizer/core/event/list/EventListFragment.java (hunk 4 above)
Comment (iamareebjamal) [reply app/src/main/java/com/eventyay/organizer/core/event/list/EventListFragment.java]: Remove

File: app/src/main/java/com/eventyay/organizer/utils/Utils.java [hunk 6]
```diff
@@ -1,12 +1,16 @@
...
+
 import com.eventyay.organizer.data.event.Event;
 
 import java.util.List;
 
 import io.reactivex.Observable;
 import io.reactivex.Single;
 
+import static com.raizlabs.android.dbflow.config.FlowManager.getContext;
+
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/utils/Utils.java]: Unneeded

Comment (tavishjain) [reply app/src/main/java/com/eventyay/organizer/utils/Utils.java]: The activity context is required in the dpToPx function to get an instance of the display metrics

Comment (tavishjain) [reply app/src/main/java/com/eventyay/organizer/utils/Utils.java]: @iamareebjamal should I pass the context as a parameter in that function ?

**Comment by kush-mish** [comment]
@tavishjain I cloned your fork and found out that on logging out and logging in again, the difference in elevation reappears.

**Screenshot**

**Comment by tavishjain** [comment]
@iamareebjamal I guess there has been too much discussion in this PR, and also a lot of vague commits by me, because I am new to open source. I am closing this PR and will be adding a new one when the issue is resolved.

**Comment by iamareebjamal** [comment]
@tavishjain This is not too much discussion, the amount of discussion normally on one PR is about 120 comments. This is because we cannot know what the maintainer is thinking and maintainers can't know what contributors are thinking. Communication is the only way this is solved. However, in order to reduce request-review cycle, contributors should ask questions and clarify what the actual issue fix should look like

**Comment by tavishjain** [comment]
I don't know why is this happening @kush-mish. Did you find any loophole that might be causing this ?

**Pull request closed without merging**